File: 卡码/代码随想录/Python/数组/03-有序数组的平方/LeetCode0977.py
#977. 有序数组的平方
from typing import List

class Solution:
	def sortedSquares(self, nums: List[int]) -> List[int]:
		res=[0 for x in range(len(nums))]
		i=0
		j=len(nums)-1
		k=j
		while i<=j:
			if nums[i]**2>nums[j]**2:
				res[k]=nums[i]**2
				i+=1
			else:
				res[k]=nums[j]**2
				j-=1
			k-=1
		return res

def main():
	nums=[-4,-1,0,3,10]
	solution=Solution()
	print(solution.sortedSquares(nums))

if __name__ == '__main__':
	main()

File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/数组/有序数组的平方/LeetCode_0977.java
package com.kurumi.codeThoughts.数组.有序数组的平方;

import java.util.Arrays;

/**
 * @Author kurumi
 * @Date 2026-07-01 下午10:59:43
 * @Description 977. 有序数组的平方
 */
public class LeetCode_0977 {
    static class Solution0977 {
        public int[] sortedSquares(int[] nums) {
            int[] res = new int[nums.length];
            int k = nums.length - 1;
            for (int i = 0, j = nums.length - 1; i <= j;) {
                if (Math.pow(nums[i], 2) > Math.pow(nums[j], 2)) {
                    res[k--] = (int) Math.pow(nums[i++], 2);
                } else {
                    res[k--] = (int) Math.pow(nums[j--], 2);
                }
            }
            return res;
        }
    }

    public static void main(String[] args) {
        Solution0977 solution0977 = new Solution0977();
        System.out.println(Arrays.toString(solution0977.sortedSquares(new int[] { -4, -1, 0, 3, 10 })));
    }
}
